packages/expo-analytics-amplitude/ios/EXAmplitude/EXAmplitude.m

```diff
@@ -80,4 +81,61 @@ - (Amplitude *)amplitudeInstance
   resolve(nil);
 }
 
+UM_EXPORT_METHOD_AS(setTrackingOptions,
+                    setTrackingOptions:(NSDictionary *)options
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject)
+{
+  AMPTrackingOptions *trackingOptions = [AMPTrackingOptions options];
+  
+  if ([options objectForKey:@"disableCarrier"]) {
```

Comment:
I think if one would provide an object like
```js
{
  "disableCarrier": false
}
```
what `NSDictionary *options` would represent would be
```objc
{
  @"disableCarrier": [NSNumber numberWithBool:false]
}
```
so the value would be a non-null pointer to a number representing a false (0) value. `if`s on such values are always positive (pointer will be `> 0`).

What I would suggest would be to:
1. change notation to a simpler `options[@"disableCarrier"]` (a cosmetic change, with regex shouldn't be that difficult and it makes the code more consistent)
2. instead of doing `if (dictionaryValue)` let's do `if ([dictionaryValue boolValue])`. If the value will be empty, it will evaluate to `false` (developer hasn't specified `disableCarrier: anything`, we don't want to disable carrier), if the value will be a number (or a boolean) it will evaluate properly and otherwise it will reject the promise with a native error (appropriate behavior).

So all in all:
```objc
if ([options[@"disableCarrier"] boolValue]) {
```